xesmf/cf.py

```diff
@@ -0,0 +1,203 @@
+                                specs_sn+'_at_')):
+                        return ds_name
+
+    # Search in axis attributes
+    if 'axis' in specs:
+        for ds_name in ds_names:
+            if (hasattr(ds[ds_name], 'axis') and
+                    ds[ds_name].axis.lower() == specs['axis'].lower()):
+                return ds_name
+
+
+def get_lon_name(ds):
+    '''
+    Get the longitude name in a `xarray.Dataset` or a `xarray.DataArray`
+
+    Parameters
+    ----------
+    ds: xarray.DataArray or xarray.Dataset
+
+    Returns
+    -------
+    str: Name of the xarray.DataArray
+    '''
+    lon_name = get_coord_name_from_specs(ds, CF_SPECS['lon'])
+    if lon_name is not None:
+        return lon_name
+    warnings.warn('longitude not found in dataset')
+
+
+def get_lat_name(ds):
+    '''
+    Get the latitude name in a `xarray.Dataset` or a `xarray.DataArray`
+
+    Parameters
+    ----------
+    ds: xarray.DataArray or xarray.Dataset
+
+    Returns
+    -------
+    str or None: Name of the xarray.DataArray
+    '''
+    lat_name = get_coord_name_from_specs(ds, CF_SPECS['lat'])
+    if lat_name is not None:
+        return lat_name
+    warnings.warn('latitude not found in dataset')
+
+
+def get_bounds_name_from_coord(ds, coord_name,
+                               suffixes=['_b', '_bnds', '_bounds']):
+    '''
+    Get the name of the bounds array from the coord array
+
+    It first searches for the 'bounds' attributes, then search
+    for names built from the suffixed coord name.
+
+    Parameters
+    ----------
+    ds: xarray.DataArray or xarray.Dataset
+    coord_name: str
+        Name of coord DataArray.
+    suffixes: list of str
+        Prefixes appended to `coord_name` to search for the bounds array name.
+
+    Returns
+    -------
+    str or None: Name of the xarray.DataArray
+    '''
+
+    # Inits
+    coord = ds[coord_name]
+
+    # From bounds attribute (CF)
+    if 'bounds' in coord.attrs:
+        bounds_name = coord.attrs['bounds'].strip()
+        if bounds_name in ds:
+            return bounds_name
+        warnings.warn('invalid bounds name: ' + bounds_name)
+
+    # From suffixed names
+    for suffix in suffixes:
+        if coord_name+suffix in ds:
+            return coord_name + suffix
+
+
+def decode_cf(ds, mapping=None):
+    '''
+    Search for longitude and latitude coordinates and bounds and rename them
+
+    Parameters
+    ----------
+    ds: xarray.DataArray or xarray.Dataset
+    mapping: None or dict
+        When a `dict` is provided, it is filled with keys that are the new
+        names and values that are the old names, so that the output dataset
+        can have its coordinates be renamed back with
+        :meth:`~xarray.Dataset.rename`.
+
+    Returns
+    -------
+    ds: xarray.DataArray or xarray.Dataset
+    '''
+
+    # Longitude
+    lon_name = get_lon_name(ds)
+    if lon_name is not None:
+
+        # Search for bounds and rename
```

Comment:
We should not have to rename coordinates.

Reply:
In fact, renaming is for internal use only, just to make the code clearer. It is possible to not rename :)